### PR TITLE
Feat: Added Chest Reward System to Activity Step

### PR DESCRIPTION
**Pull request by cloudthepain**

### CODE REVIEW

**Comment by cloudthepain** [comment]
Handles issues #170

File: IntroToCSharp/Shared/Challenges/Challenge.razor [hunk 1]
```diff
@@ -24,4 +26,8 @@
     public int Gold {get; set;} = 100;
     [Parameter]
     public int XP {get; set;} = 25;
+
+    [Parameter]
+    public StoreItem? ChestReward {get; set;} = new StoreItem("Axe", "Weapons", "Axes", 1000, "A simple, but effective axe.",
```
Comment (jcollard) [inline IntroToCSharp/Shared/Challenges/Challenge.razor:31]: Can we make the default null?

File: IntroToCSharp/Shared/Challenges/Dumb.razor [hunk 2]
```diff
@@ -0,0 +1,16 @@
...
+@code {
+    public override string ActivityType {get => "challenge"; set {}}
+
+        protected override void Claim()
+        {
+            base.Claim();
+            // TODO: Also give Item
+        }
+
+}
```
Comment (jcollard) [inline IntroToCSharp/Shared/Challenges/Dumb.razor:16]: This file probably shouldn't be included.

**Review by jcollard — CHANGES_REQUESTED** [review]
A great start! Can we make it so you can click to find out what the reward will be? Maybe when they click "Claim" it can even have a dialog pop up showing them what they've earned.

**Pull request merged**